integration-tests/src/test/java/tachyon/master/JournalCrashIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+    MasterInfo info = new MasterInfo(new InetSocketAddress(9999), journal, mExecutorsForMasterInfo,
+        mMasterTachyonConf);
+    info.init();
+
+    Assert.assertEquals(successFiles, info.listFiles(new TachyonURI(TEST_FILE_DIR), false).size());
+    for (int f = 0; f < successFiles; f ++) {
+      Assert.assertTrue(info.getFileId(new TachyonURI(TEST_FILE_DIR + f)) != -1);
+    }
+    Assert.assertEquals(successTables,
+        info.listFiles(new TachyonURI(TEST_TABLE_DIR), false).size());
+    for (int t = 0; t < successTables; t ++) {
+      Assert.assertTrue(info.getRawTableId(new TachyonURI(TEST_TABLE_DIR + t)) != -1);
+    }
+    info.stop();
+  }
+
+  private LocalTachyonClusterMultiMaster setupMultiMasterCluster() throws IOException,
+      InterruptedException {
+    // Setup the cluster with the specified ThreadGroup.
+    final LocalTachyonClusterMultiMaster cluster = new LocalTachyonClusterMultiMaster(100,
+        TEST_MASTERS, TEST_BLOCK_SIZE);
+    Thread clusterThread = new Thread(mLocalClusterThreadGroup, new Runnable() {
+      @Override public void run() {
+        try {
+          cluster.start();
+        } catch (IOException e) {
+          throw new RuntimeException(e + "\n Start Cluster Error \n" + e.getMessage(), e);
+        }
+      }
+    });
+    clusterThread.start();
+    clusterThread.join();
+    mMasterTachyonConf = cluster.getMasterTachyonConf();
+    mCreateFileThread = new ClientThread(0, cluster.getClient());
+    mCreateTableThread = new ClientThread(1, cluster.getClient());
+    mExecutorsForClient.submit(mCreateFileThread);
+    mExecutorsForClient.submit(mCreateTableThread);
+    return cluster;
+  }
+
+  private LocalTachyonCluster setupSingleMasterCluster() throws IOException, InterruptedException {
+    // Setup the cluster with the specified ThreadGroup.
+    final LocalTachyonCluster cluster = new LocalTachyonCluster(100, 100, TEST_BLOCK_SIZE);
+    Thread clusterThread = new Thread(mLocalClusterThreadGroup, new Runnable() {
+      @Override public void run() {
+        try {
+          cluster.start();
+        } catch (IOException e) {
+          throw new RuntimeException(e + "\n Start Cluster Error \n" + e.getMessage(), e);
+        }
+      }
+    });
+    clusterThread.start();
+    clusterThread.join();
+    mMasterTachyonConf = cluster.getMasterTachyonConf();
+    mCreateFileThread = new ClientThread(0, cluster.getClient());
+    mCreateTableThread = new ClientThread(1, cluster.getClient());
+    mExecutorsForClient.submit(mCreateFileThread);
+    mExecutorsForClient.submit(mCreateTableThread);
+    return cluster;
+  }
+
+  @Test
+  public void singleMasterJournalCrashIntegrationTest() throws Exception {
+    LocalTachyonCluster cluster = setupSingleMasterCluster();
+    CommonUtils.sleepMs(null, TEST_TIME_MS);
+    // Crash the cluster. Kill all the threads associated with the cluster.
+    try {
+      mLocalClusterThreadGroup.stop();
+    } catch (Error e) {
```

Comment:
I tried to ignore the "ThreadDeath" thrown by Thread.stop(). And it's already handled by overriding the "uncaughtException" in ThreadGroup, so here is redundant. I'll remove the try-catch.